### ISSUE TITLE
ArpDirection is total mess

### ISSUE THREAD

**Issue by kaikoga** [ArpEngine/ArpEngine#22]
## Why #21 fails on JS

- We are `Std.int()`ing everywhere anyway, which is wrong (because it is `UInt` in AS3) but inevitable (we use `* DEG_TO_VAL` to convert between units, which produces `Float`, and there is no `Std.uint()` which converts `Float` directly into `UInt`)
  - So `ArpDirection.value` is `Int` value in JS and fails some tests when Dynamic accessed because `UInt` conversion is bypassed
- We explicitly want it unsigned (calculated index is always >= 0) integer (to benefit from implicit wrapping), which assumption is incorrect in JS target (`haxe.Int32` must be used for that purpose, but unfortunately it is signed int32)
  - see https://haxe.org/manual/types-overflow.html

## So we must

- Hide `ArpDirectionValue` (and therefore hide `ArpDirection.new()`)
- Think `ArpDirectionValue` typing
  - Static targets
    - `ArpDirectionValue` should be `UInt`
    - Because in theory we should have no problems
  - Dynamic targets (except Neko)
    - `ArpDirectionValue` may be `UInt`
      - Undelying type of `UInt` is `Int`
    - Store as `Int32` but convert to `UInt` when necessary, if handling `UInt` seems costly
    - `Float` is not an option, all-positive values are expected
  - Neko
    - Use `UInt`, 32bit integers are already heavy
- `ArpDirection.value` should be getter (which breaks Dynamic access but its Dynamic behaviors could not be enforced so we must live with this)

## TL;DR

`ArpDirection` dependent on AS3
